src/main/scala/com/campudus/tableaux/helper/JsonUtils.scala

```diff
@@ -41,65 +41,71 @@ object JsonUtils extends LazyLogging {
-
           // optional fields
           val ordering = Try(json.getInteger("ordering").longValue()).toOption
-          val languagetype = parseMultilanguageField(json)
           val identifier = Try[Boolean](json.getBoolean("identifier")).getOrElse(false)
 
-          val countryCodes = ifContainsDo(json, "countryCodes", {
-            json =>
-              checkAllValuesOfArray[String](json.getJsonArray("countryCodes"), d => d.isInstanceOf[String] && d.matches("[A-Z]{2}|[A-Z]{3}"), "countryCodes").get
-          }).map(_.asScala.toSeq.map({ case code: String => code }))
+          // languageType or deprecated multilanguage
+          // if languageType == 'country' countryCodes must be specified
+          val languageType = parseJsonForLanguageType(json)
 
-          val di = DisplayInfos.allInfos(json)
+          // displayName and description; both multi-language objects
+          val displayInfos = DisplayInfos.allInfos(json)
 
           dbType match {
             case AttachmentType =>
-              CreateAttachmentColumn(name, ordering, identifier, di)
+              CreateAttachmentColumn(name, ordering, identifier, displayInfos)
 
             case LinkType =>
               // link specific fields
               val toName = Try(Option(json.getString("toName"))).toOption.flatten
               val singleDirection = Try[Boolean](json.getBoolean("singleDirection")).getOrElse(false)
               val toTableId = notNull(json.getLong("toTable").toLong, "toTable").get
 
-              CreateLinkColumn(name, ordering, toTableId, toName, singleDirection, identifier, di)
+              CreateLinkColumn(name, ordering, toTableId, toName, singleDirection, identifier, displayInfos)
 
             case _ =>
-              CreateSimpleColumn(name, ordering, dbType, languagetype, identifier, di, countryCodes)
+              CreateSimpleColumn(name, ordering, dbType, languageType, identifier, displayInfos)
           }
         }
     })
   } yield tuples).get
 
-  private def parseMultilanguageField(json: JsonObject): LanguageType = {
+  private def parseJsonForLanguageType(json: JsonObject): LanguageType = {
     if (json.containsKey("languageType")) {
-      val option = json.getString("languageType") match {
-        case "single" => None
-        case "language" => Some("language")
-        case "country" => Some("country")
-        case _ => throw new InvalidJsonException("Field 'languageType' should only contain 'single', 'language' or 'country'", "languagetype")
+      json.getString("languageType") match {
+        case LanguageType.LANGUAGE => MultiLanguage()
+        case LanguageType.COUNTRY =>
+          if (json.containsKey("countryCodes")) {
+            import scala.collection.JavaConverters._
+
+            val countryCodeSeq = checkAllValuesOfArray[String](json.getJsonArray("countryCodes"),
+              d => d.isInstanceOf[String] && d.matches("[A-Z]{2}|[A-Z]{3}"), "countryCodes")
```

Comment:
`[A-Z]{2,3}` should work here as well. @Zwergal told me that he changed something regarding language codes and a regular expression regarding it. Maybe check with him if there might be a problem with this regexp over here.